File: app/models/Watch.java
package models;

import com.google.appengine.api.datastore.DatastoreService;
import java.util.ArrayList;
import java.util.List;
import siena.*;
import sun.reflect.generics.reflectiveObjects.NotImplementedException;

@Table("Watch")
public class Watch extends Model {

    @Id
    public Long id;

    @Column("term")
    @Max(200) @NotNull
    public String term;

    /*
     * Instance Methods
     */

    public Watch(String term) {
        this.term = term;
    }

    public List<User> users() {
        List users = new ArrayList();
        List<UserWatch> userWatches = Model.all(UserWatch.class).
                filter("watchId", this.id).fetch();
        for (UserWatch uw : userWatches) {
            users.add(Model.all(User.class).filter("id", uw.userId).get());
        }
        return users;
    }

    /*
     * Finders
     */

    
    public static Watch findOrCreate(String term) {
        Watch w = byTerm(term);
        if (w == null) {
            w = new Watch(term);
            w.insert();
        }
        return w;
    }

    public static Watch byTerm(String term) {
        return Model.all(Watch.class).filter("term", term).get();
    }
}

File: app/models/Watch.scala
/*package models

import siena._
import reflect.BeanProperty
import java.util.ArrayList
import scala.collection.JavaConversions._

@Table("Watch")
class Watch extends Model {

  @Id
  @BeanProperty
  var id:Long = -1

  @Column(Array("term"))
  @Max(200) @NotNull
  @BeanProperty
  var term: String = null


  def users():java.util.List[User] = {
    var users = new ArrayList[User];
    var userWatches = Model.all(classOf[UserWatch]).filter("watchId", this.id).fetch
    userWatches.foreach { uw =>
      users.add(Model.all(classOf[User]).filter("id", uw.getId).get)
    }
    users
  }
}

object Watch {

  def findOrCreate(term:String):Watch = {
        var w = byTerm(term);
        if (w == null) {
          w = new Watch()
          w.term = term
          w.insert
        }
        return w
    }

  def byTerm(term: String):Watch = {
    Model.all(classOf[Watch]).filter("term", term).get
  }

  def all = {
    Model.all(classOf[Watch])
  }

}
*/
